chrome/content/zotero/xpcom/style.js

```diff
@@ -399,6 +431,72 @@ Zotero.Styles = new function() {
 			}
 		});
 	}
+	
+	/**
+	 * Populate menulist with locales
+	 */
+	this.populateLocaleList = function(menulist, prefLocale) {
+		if(!_initialized) this.init();
+		
+		var doc = menulist.ownerDocument;
+		var popup = doc.createElement('menupopup');
+		menulist.appendChild(popup);
+		
+		var desiredLocale = "";
+		var fallbackLocale = Zotero.locale;
+		
+		// Primary dialect conversion (e.g. "en" to "en-US")
+		if (Zotero.Styles.primaryDialects[prefLocale] !== undefined) {
+			prefLocale = Zotero.Styles.primaryDialects[prefLocale];
+		}
+		if (Zotero.Styles.primaryDialects[fallbackLocale] !== undefined) {
+			fallbackLocale = Zotero.Styles.primaryDialects[fallbackLocale];
+		}
+		
+		if (prefLocale) {
+			desiredLocale = prefLocale;
+		} else {
+			desiredLocale = fallbackLocale;
+		}
+		
+		var menuLocales = {};
+		var menuLocalesKeys = [];
+		var styleLocales = Zotero.Styles.locales;
+		
+		for (let locale in styleLocales) {
+			if (styleLocales[locale] !== undefined) {
+				menuLocales[locale] = styleLocales[locale];
+				menuLocalesKeys.push(locale);
+			}
+		}
+		
+		menuLocalesKeys.sort();
+		
+		if (fallbackLocale && menuLocales[fallbackLocale] === undefined) {
+			menuLocales[fallbackLocale] = fallbackLocale;
+			menuLocalesKeys.unshift(fallbackLocale);
+		}
+		if (prefLocale && menuLocales[prefLocale] === undefined) {
+			menuLocales[prefLocale] = prefLocale;
+			menuLocalesKeys.unshift(prefLocale);
+		}
```

Comment:
Do we really want to allow selecting unsupported locales here? `prefLocale` is probably never going to be anything we don't know about (in CSL), but `Zotero.locale` could be anything. I think it should just default to `en-US` if `menuLocales[fallbackLocale] == undefined` (unless Zotero and citeproc know how to deal with unsupported locales, but even then, it would be unnecessarily odd to display unformatted language codes in the list)